File: Survey2-Functional/fib.hs
-- Function: Fibonacci w/ Tail-Recursion
-- Team: Haskers
-- Author: Ryan Rosier
-- Reviewer:

fib n =
    fibo n 0 1

fibo x a b =
	if x > 0
        then fibo (x-1) b (a+b)
        else a

main = print (fib 7)

File: Survey2-Functional/fib.clj
;Fibonacci w/ Tail-Recursion
;Team: Haskers
;Author: Ryan Rosier
;Reviewer:

(defn fibo [n, a, b]
    (cond (> n 0)
        (fibo (- n 1) b (+ a b))
        :else a
    )
)

(print (fibo 6 0 1))
